x-pack/plugins/reporting/export_types/printable_pdf/server/lib/layouts/optimized_layout.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import path from 'path';
+import {
+  ConfigEntry,
+  configObject,
+  kbn_server,
+  ViewWidth,
+  ViewWidthHeight,
+  ViewZoomWidthHeight,
+} from '../../../../../../../../src/server/index';
+import { Layout } from './layout';
+
+type EvalArgs = any[];
+
+interface EvaluateOptions {
+  // Fn is a function in string form to avoid tslint from auto formatting it into a version not
+  // underfood by transform_fn safeWrap.
+  fn: ((evalArgs: EvalArgs) => any) | string;
+  args: EvalArgs; // Arguments to be passed into the function defined by fn.
+}
+
+interface BrowserClient {
+  evaluate: (evaluateOptions: EvaluateOptions) => void;
+}
+
+export class Optimizedlayout extends Layout {
+  public groupCount: number = 2;
+
+  public selectors = {
+    screenshot: '[data-shared-item]',
+    renderComplete: '[data-shared-item]',
+    itemsCountAttribute: 'data-shared-items-count',
+    timefilterFromAttribute: 'data-shared-timefilter-from',
+    timefilterToAttribute: 'data-shared-timefilter-to',
+  };
+
+  constructor(server: kbn_server, id: string) {
+    super(id, server);
+  }
+
+  get config(): configObject {
+    return this.server.config();
+  }
+
+  get captureConfig(): ConfigEntry {
```

Comment:
I'm not a fan of the `get ` syntax, and I think it adds unnecessary execution cycles.  Can just make `captureConfig` a property on the class (i think const since it never changes), same with most of the other ones.  e.g.:

```
 private const captureConfig: CaptureConfig; 
 constructor(server: kbn_server, id: string) {
    super(id); //(getting rid of server object into base class re: other comment)
    this.captureConfig = server.config().get('xpack.reporting.capture')
  }
```

actually... if the server object is only used to grab that capture config, you might not need to keep a reference around to the server object at all.  This is more in line with the original code, which also did not keep a reference to the server object in the returned object.

Actually, can just pass in captureConfig object to the constructor and do the grabbing of it in the factory function.  Would make this class easier to test.

```
 private const captureConfig: CaptureConfig; 
 constructor(captureConfig: CaptureConfig, id: string) {
    super(id); //(getting rid of server object into base class re: other comment)
    this.captureConfig = captureConfig;
  }
```